### PR TITLE
Fix invest errors

### PR DESCRIPTION
**Pull request by ztolley**
Fixed 2 errors:

1: Fixed repopulating the contact and business activity fields when saving an investment summary with errors. Added unit tests to confirm.
2: Fixed error not showing error messages for business activities and contacts when saving

errors

### CODE REVIEW

File: src/apps/investment-projects/views/_details-form.njk [hunk 1]
```diff
@@ -90,12 +90,16 @@
     value: form.state['sector']
   }) }}
 
-  <fieldset class="c-form-fieldset">
+  <fieldset class="c-form-fieldset {{ 'has-error' if form.errors.messages['business_activities'] }}">
```
Comment (teneightfive) [inline src/apps/investment-projects/views/_details-form.njk]: Wonder if would be easier to use the `FormGroup` macro here with `fieldset` as the element and would mean we don't have this code duplicated?

Comment (ztolley) [reply src/apps/investment-projects/views/_details-form.njk]: what would that look like?

**Comment by feedmypixel** [comment]
@ztolley why can't you just use the form errors work we have?

File: src/apps/investment-projects/views/_details-form.njk (hunk 1 above)
Comment (ztolley) [reply src/apps/investment-projects/views/_details-form.njk]: Tried using formgroup macro but ran into issues with styling due to errors being shown within the legend. Any changes to formgroup would have an adverse effect on Omis.

Re. use of dot notation, the whole template uses [''] so sticking with that for consistency.

**Comment by ztolley** [comment]
@feedmypixel the form errors get rendered in the legend which totally throw the layout.

**Comment by feedmypixel** [comment]
@ztolley can you adda gif? Its hard to work out what the actual problem is here

**Pull request merged**